File: Class/Practise 4.1/src/Pizza.cpp
#include "Pizza.h"
#include <print>
#include <iostream>
#include <numeric>

void Pizza::addIngredient(const Ingredient& ingredient) {
    ingredients.push_back(ingredient);
}

void Pizza::printPizza() const {
    double totalCost = 0;
    std::println("Your pizza ingredients:");
    for (const auto& ingredient : ingredients) {
        std::println("- {} (${})", ingredient.getName(), ingredient.getCost());
        totalCost += ingredient.getCost();
    }
    std::println("Total cost: ${}", totalCost);
}

std::vector<Ingredient> Pizza::getIngredients() const {
    return ingredients;
}

double Pizza::getTotalCost() const {
    return std::accumulate(ingredients.begin(), ingredients.end(), 0.0, [](double sum, const Ingredient& ingredient) {
        return sum + ingredient.getCost();
    });
}

File: Home/Home 4.1/Pizza.swift
import Foundation

class Pizza {
    private var ingredients: [Ingredient] = []

    func addIngredient(_ ingredient: Ingredient) {
        ingredients.append(ingredient)
    }

    func printPizza() {
        var totalCost = 0.0
        print("Your pizza ingredients:")
        for ingredient in ingredients {
            print("- \(ingredient.name) ($\(ingredient.cost))")
            totalCost += ingredient.cost
        }
        print("Total cost: $\(totalCost)")
    }

    func getIngredients() -> [Ingredient] {
        return ingredients
    }

    func getTotalCost() -> Double {
        return ingredients.reduce(0) { $0 + $1.cost }
    }
}
